src/dxtbx/model/detector.h

```diff
@@ -763,6 +763,15 @@ namespace dxtbx { namespace model {
         (*this)[i].rotate_around_origin(axis, angle);
       }
     }
+  
+    bool has_projection_2d(){
+      for (std::size_t i = 0; i < size(); ++i){
+        if (!(*this)[i].has_projection_2d()){
+          return false;
+        }
+      }
+      return true;
+    }
```

Comment:
Do we need an explicit `Detector.has_projection_2d()` function, particularly if we're making the suggested changes below to remove `Panel.has_projection_2d()` and allow `Panel.get_projection_2d()` to optionally return `None`. Then you would just be able to do e.g. `all([p.get_projection_2d() for p in detector])` in Python, but with more control over what to do if only _some_ of the panels have a 2d projection.